src/Services/IAlgoliaIndexService.cs

```diff
@@ -7,10 +8,27 @@ namespace Kentico.Xperience.AlgoliaSearch.Services
     /// </summary>
     public interface IAlgoliaIndexService
     {
+        /// <summary>
+        /// Gets the desired name for the specified index. This can be used to append text to the index
+        /// names based on the current environment, e.g. "DEV-Xperience" and "PROD-Xperience."
+        /// </summary>
+        /// <param name="indexName">The original name of the index.</param>
+        /// <returns>The final name of the index.</returns>
+        string GetIndexName(string indexName);
```

Comment:
This is really confusing. You have to/or not? call this method before the InitializeIndex method (You have changed all the examples). I'm not sure wheter it's mandatory or not to do that. I don't need to call this if I don't want to change the name by custom code, right?

Should I call it before SetIndexSettings or not? I know the answer, but I want to show the confusing parts. 

I would prefer to have a dedicated service for  name modification. It will be called in all places internally. No one will be forced to call it for default services. The only places where it will be used directly is algolia API calls (the modification class can be called, so there will be only one place where the name is changed - similar to current approach)

Reply:
The custom name needs to be exposed for the client JS integration, so unfortunately it can't be completely internal...

Reply:
I've added `InitializedIndex` in https://github.com/Kentico/xperience-algolia/pull/24/commits/888a501ff25850ecc57517fdde90dfed3deed90d, what do you guys think of this approach?

Reply:
> This is also really useful since the ISearchIndex interface doesn't expose the name, now customers can easily get the name at any time.

That's what I don't like about the current approach. Currently, an additional call is needed to get name of the index that has already been defined. The API signatures consume a string parameter, and nothing guide you to use that call.

The non-attribute registration  suggested by Sean, seems to be a minimalistic and workable solution. The name will be defined within the configuration phase based on any variable. We don't need to remove the attribute registration (because of backward compatibility), it can be an alternative way to register the index model.

Reply:
I guess I'm confused about what the configuration approach involves. Can you guys describe that in greater detail?

Do you mean that the name is generated [during registration](https://github.com/Kentico/xperience-algolia/blob/master/src/Services/Implementations/DefaultAlgoliaRegistrationService.cs#L246) (auto-generated from the type) and this is where developers can customize the name if needed? Then, the custom/non-custom name will be stored in the [RegisteredIndexes](https://github.com/Kentico/xperience-algolia/blob/master/src/Services/Implementations/DefaultAlgoliaRegistrationService.cs#L37) and no further logic would be needed to get the name... _except_ looping through `RegisteredIndexes` to find an index with the desired type (a helper method could be made to assist).

Reply:
I'm thinking about alternative way how to register the model. Something like this:
```csharp
services.AddAlgolia(configuration, register =>
	{
		register.Add<AlgoliaSiteSearchModel>(AlgoliaSiteSearchModel.IndexName, /* optional */ new[] { "SiteA" } );
	});
```

If you need to change the name based on the env. variable you can do it here. Nothing like an auto-generated name from the type is needed...

Reply:
I see, it needs to be registered in the CMS project as well though. That could be done in a custom module. @seangwright what do you think?